core/src/test/java/com/radixdlt/rev2/LedgerProofsGcTest.java

```diff
@@ -0,0 +1,200 @@
+                        REV2ProposerConfig.transactionGenerator(
+                            new MbTransactionGenerator(NetworkDefinition.INT_TEST_NET), 3),
+                        false,
+                        StateHashTreeGcConfig.forTesting(),
+                        new LedgerProofsGcConfig(
+                            UInt32.fromNonNegativeInt(GC_INTERVAL_SEC),
+                            UInt64.fromNonNegativeLong(mostRecentFullResolutionEpochCount)),
+                        false))));
+  }
+
+  @Test
+  public void node_prunes_non_critical_proofs_from_configured_old_enough_epochs() {
+    // Arrange: configure ledger proof gc to keep 2 most recent completed epochs not pruned
+    var mostRecentFullResolutionEpochCount = 2;
+    /// ... 6 rounds * 3 transactions * 1 MB = 18MB; Limit is 12 MB - so we require 2 proofs
+    var epochLength = 6;
+    try (var test = createTest(mostRecentFullResolutionEpochCount, epochLength)) {
+      test.startAllNodes();
+
+      // Act: advance to epoch 5
+      test.runUntilState(NodesPredicate.nodeAt(0, NodePredicate.atOrOverEpoch(5)), 10000);
+
+      // Assert: after an async GC, we expect certain number of proofs in each epoch:
+      Awaitility.await()
+          .atMost(2 * GC_INTERVAL_SEC, TimeUnit.SECONDS)
+          .untilAsserted(
+              () -> {
+                var stateReader = test.getInstance(0, TestStateReader.class);
+                // - The epoch 5 has just started, hence it has no proofs
+                assertThat(stateReader.countProofsWithinEpoch(5)).isEqualTo(0);
+                // - The 2 most recent completed epochs contain all their proofs (i.e. not pruned)
+                assertThat(stateReader.countProofsWithinEpoch(4)).isEqualTo(epochLength);
+                assertThat(stateReader.countProofsWithinEpoch(3)).isEqualTo(epochLength);
+                // - This epoch was pruned, and it has 2 proofs (1 would not fit the 12MB limit)
+                assertThat(stateReader.countProofsWithinEpoch(2)).isEqualTo(2);
+              });
+
+      // Follow-up: Advance one more epoch
+      test.runUntilState(NodesPredicate.nodeAt(0, NodePredicate.atOrOverEpoch(6)), 10000);
+
+      // Assert: after an async GC...
+      Awaitility.await()
+          .atMost(2 * GC_INTERVAL_SEC, TimeUnit.SECONDS)
+          .untilAsserted(
+              () -> {
+                // ... the "pruning window" has progressed (i.e. now epoch 3 got pruned)
+                var stateReader = test.getInstance(0, TestStateReader.class);
+                assertThat(stateReader.countProofsWithinEpoch(6)).isEqualTo(0);
+                assertThat(stateReader.countProofsWithinEpoch(5)).isEqualTo(epochLength);
+                assertThat(stateReader.countProofsWithinEpoch(4)).isEqualTo(epochLength);
+                assertThat(stateReader.countProofsWithinEpoch(3)).isEqualTo(2);
+                assertThat(stateReader.countProofsWithinEpoch(2)).isEqualTo(2);
+              });
```

Comment:
Might be nice to include a full node, but only turn it on now, and ensure that it can sync up to epoch 6 :D

Reply:
that's a good separate test 👍